spec/classes/puppet_server_puppetserver_spec.rb

```diff
@@ -276,10 +277,33 @@
         end
       end
 
-      describe 'server_trusted_agents' do
-        context 'when set' do
+      describe 'puppetlabs v3 catalog from agents rule' do
+        context 'when server_trusted_agents set' do
           let(:params) { super().merge(server_puppetserver_trusted_agents: ['jenkins', 'octocatalog-diff']) }
-          it { should contain_file(auth_conf).with_content(/^            allow: \["jenkins", "octocatalog-diff", "\$1"\]$/) }
+          it do
+            content = catalogue.resource('file', auth_conf).send(:parameters)[:content]
+            rules = Hocon.parse(content)['authorization']['rules']
+            rule = rules.find {|rule| rule['name'] == 'puppetlabs v3 catalog from agents' }
+            expect(rule['allow']).to eq(['jenkins', 'octocatalog-diff', '$1'])
+          end
+        end
+        context 'with server_trusted_certificate_extensions' do
```

Comment:
I always like to visually separate context blocks with an empty line (GH renders an ugly diff)
```suggestion

        context 'with server_trusted_certificate_extensions' do
```